### PR TITLE
add rusticl to the site 

### PR DESCRIPTION
**Pull request by norablackcat**
This splits Clover and Rusticl into two different sections and handles them independently 

Parsing based on this merge req https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/23055

closes: https://github.com/MightyCreak/mesamatrix/issues/269

### CODE REVIEW

**Comment by norablackcat** [comment]
Everything should be ready now baring any changes from reviewers on mesa

File: src/Leaderboard/Leaderboard.php [hunk 1]
```diff
@@ -56,7 +56,9 @@ public function load(\SimpleXMLElement $xmlApi)
             } elseif (
                 $a->getName() === Constants::GL_NAME ||
                 $a->getName() === Constants::VK_NAME ||
-                $a->getName() === Constants::OPENCL_NAME
+                $a->getName() === Constants::CLOVER_OPENCL_NAME ||
+                $a->getName() === Constants::RUSTICL_OPENCL_NAME
+
```
Comment (MightyCreak) [inline src/Leaderboard/Leaderboard.php:61]: Just an extra newline that can be removed

**Review by MightyCreak — CHANGES_REQUESTED** [review]
Just a tiny change, but LGTM, thanks!

**Comment by norablackcat** [comment]
Upstream has been merged :crab:

**Comment by illwieckz** [comment]
I tested this, that looks good! 🙂️

Why not listing RustiCL before Clover? I know RustiCl is still experimental, but Clover is not less experimental, just experimental since more time, and is less complete.

**Comment by zboszor** [comment]
> Why not listing RustiCL before Clover? I know RustiCl is still experimental, but Clover is not less experimental, just experimental since more time, and is less complete.

I think it's not really interesting. But it fits alphabetical order: Clover < Rusticl.

**Comment by norablackcat** [comment]
need to update this to add r600 as it's been merged so do not merge yet. 



> > Why not listing RustiCL before Clover? I know RustiCl is still experimental, but Clover is not less experimental, just experimental since more time, and is less complete.
> 
> I think it's not really interesting. But it fits alphabetical order: Clover < Rusticl.

Yeah that's the order I did it in initially although it's probably better to put rusticl above because it supports more and is actually maintained

**Comment by norablackcat** [comment]
Changed to display rusticl first and added the possible drivers rusticl will support in the future.

Need to double check with Karol to make sure those drivers are correct

**Comment by norablackcat** [comment]
It was decided that we are not going to list unsupported drivers and just update the list later when the need arises for some wip driver support

this should be ready for merge

**Comment by MightyCreak** [comment]
Hmmm there are some issues with the tests as well, I'm taking a look at it right now.

**Review by MightyCreak — APPROVED** [review]
Pushed some lint and tests fixes, but it's all good now!

**Pull request merged**